railties/lib/rails/tasks/yarn.rake

```diff
@@ -16,7 +16,9 @@ namespace :yarn do
         "--immutable"
       end
 
-    system({ "NODE_ENV" => node_env }, "#{RbConfig.ruby} \"#{Rails.root}/bin/yarn\" install #{yarn_flags}")
+    unless system({ "NODE_ENV" => node_env }, "#{RbConfig.ruby} \"#{Rails.root}/bin/yarn\" install #{yarn_flags}")
```

Comment:
The `system` method has the argument `exception: true` to raise an error on a non-zero exist status. Would it make sense to use that for a familiar error message rather than inventing a new one?

https://ruby-doc.org/core-3.0.0/Kernel.html#method-i-system

Reply:
Good suggestion. I'd like to use the `:exception` option, now that I'm aware of it, but rails still supports ruby 2.5, right? AFAICT, the option was not added until 2.7.

Reply:
The next release will be Rails 7.0, and will require Ruby 2.7 or greater (https://github.com/rails/rails/commit/6487836af8f50648a9b30ce61864c827132e5592).